crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -590,11 +596,20 @@ impl AuthorityPerEpochStore {
         Ok(self.tables.next_shared_object_versions.multi_get(ids)?)
     }
 
-    pub fn get_last_checkpoint_boundary(&self) -> (u64, Option<u64>) {
-        match self.tables.checkpoint_boundary.iter().skip_to_last().next() {
-            Some((idx, height)) => (idx, Some(height)),
-            None => (0, None),
-        }
+    pub fn get_last_checkpoint_boundary(&self) -> Result<(u64, Option<u64>), TypedStoreError> {
+        Ok(
+            match self
+                .tables
+                .checkpoint_boundary
+                .safe_iter()
```

Comment:
Since we are doing this `iter()` migration, I hope we can also consider having methods like `iter_from(key)` and `iter_to_last()`, that seeks only once in rocksdb. The current way of `.iter().skip_to(...)` seeks twice, first to the beginning of the table then to the actual key or end of the table.